x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/connectors/create_connector/create_connector.tsx

```diff
@@ -0,0 +1,498 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState } from 'react';
+
+// import { useLocation } from 'react-router-dom';
+
+import { css } from '@emotion/react';
+
+import { useValues, useActions } from 'kea';
+
+import {
+  EuiBadge,
+  EuiButtonEmpty,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiIcon,
+  EuiLink,
+  EuiPanel,
+  EuiSpacer,
+  EuiSteps,
+  EuiSuperSelect,
+  EuiText,
+  useEuiTheme,
+} from '@elastic/eui';
+
+import { EuiStepInterface } from '@elastic/eui/src/components/steps/step';
+import { i18n } from '@kbn/i18n';
+// import { FormattedMessage } from '@kbn/i18n-react';
+
+import { KibanaLogic } from '../../../../shared/kibana';
+import { ConnectorViewLogic } from '../../connector_detail/connector_view_logic';
+import { EnterpriseSearchContentPageTemplate } from '../../layout';
+import { connectorsBreadcrumbs } from '../connectors';
+
+import connectorsBackgroundImage from './assets/connector_logos_comp.png';
+
+import { ConfigurationStep } from './configuration_step';
+import { DeploymentStep } from './deployment_step';
+import { FinishUpStep } from './finish_up_step';
+import { StartStep } from './start_step';
+
+export const CreateConnector: React.FC = () => {
+  const { euiTheme } = useEuiTheme();
+  const [selfManaged, setSelfManaged] = useState(false);
+  const { connectorTypes } = useValues(KibanaLogic);
+  const allConnectors = connectorTypes.sort((a, b) => a.name.localeCompare(b.name)); // alphabetically ordered
+  const { fetchConnector } = useActions(ConnectorViewLogic);
+  const [syncing, setSyncing] = useState(false);
+  const { connector } = useValues(ConnectorViewLogic);
+  useEffect(() => {
+    fetchConnector({ connectorId: 'eIwou5AB7hZjs4c7Qmm4' });
+  }, []);
+  const [startStepStatus, setStartStepStatus] = useState<
+    | 'current'
```

Comment:
You can reuse Eui's stepstatus types instead of writing them one by one.
I think it is called `EuiStepStatus`, you can extend it of course.

Reply:
Done